### DISCUSSION TITLE
Version 2 of mini-rx-store

### DISCUSSION THREAD

**Discussion by spierala** [spierala/mini-rx-store#19] [Show and tell]
Currently we are  working on version 2 of mini-rx-store.
New features are coming. There are also breaking changes.

Changes till now:
- MetaReducers: RootMetaReducers, FeatureMetaReducers
- New LoggerExtension (uses root meta reducer unter the hood)
- New ImmutableExtension (uses root meta reducer unter the hood)
- BREAKING CHANGE: no settings anymore for logging, use LoggerExtension instead
- BREAKING CHANGE: Feature.createEffect: do not return new state anymore, simply call another setState function. Usage is heavily inspired by the ngrx/component store effect method: https://ngrx.io/guide/component-store/effect#effect-method
Usage becomes simpler, just use tap and catchError to handle the effect result. Code to update state can be reused more easily when using setState functions. 
- Add more Angular friendly API: StoreModule.forRoot(), StoreModule.forFeature, EffectsModule.register, DI support: private store: Store, private actions: Actions
- BREAKING change: Store -> store (get hold of the store instance with `configureStore({})`)
- BREAKING change: Feature -> FeatureStore
- Dispatch setState Action only to the features own reducer (for performance)
- Added support for undo `store.dispatch(undo(someAction))`. FeatureStore can use `const action = this.setState(); this.undo(action));`
 
What do you think of the changes till now?
Let us know if you have ideas for features that you wish to see in mini-rx-store@2

**Comment by hansschenker** [comment]
I am seeing rx-mini-store growing the size of NgRx - the big brother of rx-mini-store!
Do you want to stay mini or just become another NgRx?

**Reply by spierala** [reply]
Funny that you describe NgRx as the big brother of MiniRx... I also thought of that comparison :)

MiniRx is still mini, but indeed it can do almost the same as NgRx/store/effects.
If you look at the source code of MiniRx you will find that there is much less code involved than in NgRx (for Store and Effects). 
The source code which is basically the implementation of Store and Effects in MiniRx:
https://github.com/spierala/mini-rx-store/blob/master/projects/mini-rx-store/src/lib/store-core.ts
It looks still quite minimalistic to me.

MiniRx will never try to support every edge case feature as NgRx does, so it will stay mini, even if it can do a lot. 
MiniRx is meant to focus on the core features of a good state management lib and keep the implementation minimalistic and simple.

**Comment by spierala** [comment]
mini-rx-store@2-beta is out! :)  See ChangeLog for all changes which went into Version 2: https://github.com/spierala/mini-rx-store/blob/master/projects/mini-rx-store/CHANGELOG.md

**Comment by spierala** [comment]
working on nice docs currently:
https://spierala.github.io/mini-rx-store

**Comment by spierala** [comment]
Version 2 is published :)
